components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java

```diff
@@ -381,10 +387,15 @@ protected boolean handlePostAuthentication(Event event, String userName, UserSto
                         unlockTimePropertyValue = (long) (unlockTimePropertyValue * 1000 * 60 * Math.pow
                                 (unlockTimeRatio, currentFailedLoginLockouts));
                         unlockTime = System.currentTimeMillis() + unlockTimePropertyValue;
+                        IdentityUtil.threadLocalProperties.get().put(AccountConstants.USER_LOCKED_DURATION,
```

Comment:
these needs to be properly cleared. We can't guarantee that it will be cleared in handlePostSetUserClaimValues method. 

There can be situations that won't call. In that case, the thread locals will remain and use in a different thread. That may cause unnecessary flows.